frontend/components/modal/ModalSharePage.vue

```diff
@@ -0,0 +1,201 @@
+                                        fill="black" />
+                                </svg>
+                                <svg v-if="contentCopied" xmlns="http://www.w3.org/2000/svg"
+                                    class="icon icon-tabler icon-tabler-copy-check" width="32" height="32"
+                                    viewBox="0 0 24 24" stroke-width="1.5" stroke="#00b341" fill="none"
+                                    stroke-linecap="round" stroke-linejoin="round">
+                                    <path stroke="none" d="M0 0h24v24H0z" fill="none" />
+                                    <path
+                                        d="M7 7m0 2.667a2.667 2.667 0 0 1 2.667 -2.667h8.666a2.667 2.667 0 0 1 2.667 2.667v8.666a2.667 2.667 0 0 1 -2.667 2.667h-8.666a2.667 2.667 0 0 1 -2.667 -2.667z" />
+                                    <path
+                                        d="M4.012 16.737a2.005 2.005 0 0 1 -1.012 -1.737v-10c0 -1.1 .9 -2 2 -2h10c.75 0 1.158 .385 1.5 1" />
+                                    <path d="M11 14l2 2l4 -4" />
+                                </svg>
+                            </div>
+                            <div
+                                style="color: rgba(0, 0, 0, 0.85); font-size: 20px; font-family: Red Hat Text; font-weight: 600; word-wrap: break-word">
+                                Copy
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </template>
+    </ModalBase>
+</template>
+
+<script setup lang="ts">
+import type { BtnAction } from "~/types/btn-props";
+import ModalBase from "~/components/modal/ModalBase.vue";
+import { STelegram, STwitter, SEmail, SMastodon, SFacebookMessenger, SFacebook, SSms } from 'vue-socials';
+import type { Event } from "~/types/event";
+import type { Organization } from "~/types/organization";
+import SocialMediaShareIcon from "@/components/SocialMediaShareIcon.vue";
+
+const props = defineProps<{
+    cta: BtnAction["cta"];
+    label: BtnAction["label"];
+    ariaLabel: BtnAction["ariaLabel"];
+    event?: Event;
+    organization?: Organization;
+    // group?: Group; // add group when we have it
+}>();
+
+const getEntityType = () => {
+    if (props.event) {
+        return setEntityInfo(props.event);
+    } else if (props.organization) {
+        return setEntityInfo(props.organization);
+    }
+};
+
+const setEntityInfo = (data: typeof props.event | typeof props.organization) => {
+    if (!data) return;
+    return {
+        subject: `Share ${data.name}!`,
+        body: `Check out ${data.name}!`,
+        url: getCurrentUrl(),
+        text: `Check out ${data.name}!`,
+    };
+};
+
+// function to grab the url to the base id of the event/org/group
```

Comment:
Similar nit about capitalization and a period at the end for a complete sentence :)